en/concept/flight_modes.md

```diff
@@ -1,111 +1,215 @@
+From an external perspective (MAVLink/GCS), the origin of a mode is indistinguishable.
 
-The sections below provide an overview of the modes, followed by a [flight mode evaluation diagram](#flight-mode-evaluation-diagram) that shows the conditions under which PX4 will transition into a new mode.
+This topic links to documentation for the supported modes, compares PX4 FC and ROS2 modes, provides implementation hints, and provides an overview of PX4 modes can be used with MAVLink.
 
-::: info
-User-facing flight mode documentation can be found in:
+## Supported Flight Modes
+
+Not all flight modes are available (or makes sense), on all vehicle types, and some modes behave differently on different vehicle types.
+
+Flight mode documentation for the modes running on the flight controller are listed below:
 
 - [Flight Modes (Multicopter)](../flight_modes_mc/index.md)
 - [Flight Modes (Fixed-Wing)](../flight_modes_fw/index.md)
 - [Flight Modes (VTOL)](../flight_modes_vtol/index.md)
 - [Drive Modes (Differential Rover)](../flight_modes_rover/differential.md)
 - [Drive Modes (Ackermann Rover)](../flight_modes_rover/ackermann.md)
+- [Basic Configuration > Flight Modes](../config/flight_mode.md)
+
+::: info
+At time of writing there are no PX4 ROS 2 flight modes that are considered "supported as an integral part of the PX4 open source offering".
+:::
+
+## PX4 FC Modes vs PX4 ROS2 Flight Modes
+
+With some exceptions it is possible to write a mode in either the FC and companion computer.
+The main considerations are listed below.
+
+ROS 2 modes cannot be used in the following cases:
+
+- Modes that need to run on vehicles that don't have a companion computer.
+- Modes that require low-level access, strict timing, and/or high update rate requirements.
+  For example, a multicopter mode that implements direct motor control.
+- Safety critical modes, such as [Return mode](../flight_modes_mc/return.md).
+- When you can't use ROS (for any reason).
+
+PX4 ROS 2 modes should be considered for all other cases.
+They have the following benefits:
+
+- Easier to implement as there is no need to deal with low-level embedded constraints and requirements (such as restricted stack sizes).
+- Easier to maintain as the integration API is small, well defined, and stable (porting custom PX4 modes on the flight controller between PX4 versions is much harder).
+- Crashing the app does not crash the vehicle.
+- They can override existing modes to provide more advanced features.
+  You can even override a safety-critical mode with a better versions: if the ROS 2 mode crashes the original mode will be engaged.
+- High-level functionality is available, including a better-feature programming environment, and many useful Linux and ROS libraries.
+- More available compute to do more advanced processing (e.g. computer vision).
+
+Note that there are some negatives to keep in mind:
```

Comment:
I think we should open an issue where we track these three bullet points and link to this instead of listing them here where they will likely get stale over time

Reply:
@mrpollo I propose you do that - the point about the control interface being experimental is relevant here but probably wouldn't go in the same issue. We do need to talk to @bkueng about what is needed to get it to the point it is considered "stable".

I did create this one just for the CI:
- https://github.com/PX4/PX4-Autopilot/issues/24363